File: c#/listener.cs
[HttpPost]
public async Task<IActionResult> ReceiveWebhook()
{
    const string TOKEN_CORRETO = "TOKEN_DO_CLIENTE_EXEMPLO_ABC123"; // Token individual do cliente

    if (!Request.Headers.TryGetValue("X-Webhook-Token", out var token) || token != TOKEN_CORRETO)
        return StatusCode(403, "Acesso negado!");

    using var reader = new StreamReader(Request.Body);
    var body = await reader.ReadToEndAsync();

    System.IO.File.AppendAllText("webhook_log.txt", $"---- {DateTime.Now} ----\nRAW: {body}\n------------------------\n");

    // Enviar email com MailKit ou SMTP padrão aqui...

    return Ok("OK");
}

File: java/listener.js
// Token individual desse cliente
private static final String TOKEN_CORRETO = "TOKEN_DO_CLIENTE_EXEMPLO_ABC123";

@PostMapping("/")
public ResponseEntity<String> receiveWebhook(
    @RequestHeader(value="X-Webhook-Token", required=false) String token,
    @RequestBody String payload) {
    if (!TOKEN_CORRETO.equals(token)) {
        return new ResponseEntity<>("Acesso negado!", HttpStatus.FORBIDDEN);
    }
    // Gravar log (simples)
    try (FileWriter fw = new FileWriter("webhook_log.txt", true)) {
        fw.write("---- " + LocalDateTime.now() + " ----\nRAW: " + payload + "\n------------------------\n");
    } catch (Exception e) { /* trate o erro */ }
    // Enviar email com JavaMail/Spring Email (exemplo não incluído por brevidade)
    return ResponseEntity.ok("OK");
}
